File: bindings/kotlin/examples/PackageEvents.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.Direction
import iota_sdk.EventFilter
import iota_sdk.GraphQlClient
import iota_sdk.PaginationFilter
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    try {
        val client = GraphQlClient.newTestnet()

        val events =
            client.events(
                filter =
                    EventFilter(
                        eventType =
                            "0x7fff6e95f385349bec98d17121ab2bfa3e134f2f0b1ccefc270313415f7835ea::registry::NameRecordAddedEvent"
                    ),
                paginationFilter = PaginationFilter(direction = Direction.FORWARD, limit = 10),
            )

        for (event in events.data) {
            // Sender and module are optional: some events (such as system- or
            // genesis-emitted ones) carry neither.
            println("Type: ${event.moveType}")
            println("Sender: ${event.sender?.toHex() ?: "none"}")
            println("Module: ${event.module ?: "none"}")
            println("JSON: ${event.json}")
        }
    } catch (e: Exception) {
        e.printStackTrace()
        kotlin.system.exitProcess(1)
    }
}

File: bindings/swift/examples/package_events.swift
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import IotaSDK

@main
struct PackageEventsExample {
  static func main() async throws {
    let client = GraphQlClient.newTestnet()

    let events = try await client.events(
      filter: EventFilter(
        eventType:
          "0x7aec8176867a0c8d2803d758ebf98226d301ef0f00393879ea718f6bd1554f16::registry::NameRecordAddedEvent"
      ),
      paginationFilter: PaginationFilter(direction: Direction.forward, limit: 10)
    )

    for event in events.data {
      print("Type: \(event.type)")
      print("Sender: \(event.sender.toHex())")
      print("Module: \(event.module)")
      print("JSON: \(event.json)")
    }
  }
}
